hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/index/ScheduleIndexActionExecutor.java

```diff
@@ -107,7 +107,7 @@ public Option<HoodieIndexPlan> execute() {
             .orElseThrow(() -> new HoodieIndexException(String.format("Could not get metadata writer to initialize filegroups for indexing for instant: %s", instantTime)));
         if (!finalPartitionsToIndex.get(0).getPartitionPath().equals(MetadataPartitionType.FILES.getPartitionPath())) {
           // initialize metadata partition only if not for FILES partition.
-          metadataWriter.initializeMetadataPartitions(table.getMetaClient(), finalPartitionsToIndex, indexInstant.getTimestamp());
+          metadataWriter.initializeMetadataPartitions(table.getMetaClient(), finalPartitionsToIndex, indexUptoInstant.get().getTimestamp());
```

Comment:
Hey @nsivabalan , could u take a look at this line of change? The relevant change is in `HoodieAppendHandle`, where we may override instant time with the one from `currentLocation` in record.

Reply:
@YuweiXiao Let's create a JIRA and put up a separate PR for this fix.